### PR TITLE
47019_Employee_List_Screen_Load

### PR DESCRIPTION
**Pull request by khalidmahmud**

### CODE REVIEW

File: hrm_2/EmployeeListViewController.m [hunk 1]
```diff
@@ -50,7 +54,35 @@ - (UITableViewCell *)tableView:(UITableView *)employeeTableView cellForRowAtInde
...
+
+- (NSArray *)getEmployeeNames {
+    AppDelegate *appdelegate = [[UIApplication sharedApplication] delegate];
+    NSManagedObjectContext *context = [appdelegate managedObjectContext];
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"Employee"];
+    fetchRequest.fetchBatchSize=30;
+    fetchRequest.fetchLimit=3000;
+    NSString *departmentName = @"";
+    fetchRequest.predicate =
+    [NSPredicate predicateWithFormat:@"departmentOfEmployee.departmantName==%@",departmentName];
```
Comment (nguyenvandungb) [inline hrm_2/EmployeeListViewController.m]: Better if you change like this :  departmentOfEmployee.departmantName == %@

**Comment by nguyenvandungb** [comment]
OK

File: hrm_2/EmployeeListViewController.m [hunk 2]
```diff
@@ -50,7 +54,35 @@ - (UITableViewCell *)tableView:(UITableView *)employeeTableView cellForRowAtInde
...
     return cell;
 }
 
+#pragma mark - fetch Employee
+
+- (NSArray *)getEmployeeNames {
+    AppDelegate *appdelegate = [[UIApplication sharedApplication] delegate];
+    NSManagedObjectContext *context = [appdelegate managedObjectContext];
+    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"Employee"];
+    fetchRequest.fetchBatchSize=30;
```
Comment (TequilarTomsk) [inline hrm_2/EmployeeListViewController.m]: fetchRequest.fetchBatchSize=30; please add " "

File: hrm_2/EmployeeListViewController.m [hunk 3]
```diff
@@ -50,7 +54,35 @@ - (UITableViewCell *)tableView:(UITableView *)employeeTableView cellForRowAtInde
...
+    fetchRequest.fetchLimit=3000;
```
Comment (TequilarTomsk) [inline hrm_2/EmployeeListViewController.m]: add " "

File: hrm_2/EmployeeListViewController.m [hunk 4]
```diff
@@ -50,7 +54,35 @@ - (UITableViewCell *)tableView:(UITableView *)employeeTableView cellForRowAtInde
...
+    NSString *departmentName = @"";
+    fetchRequest.predicate =
+    [NSPredicate predicateWithFormat:@"departmentOfEmployee.departmantName == %@",departmentName];
+    
+    NSSortDescriptor *sortDescriptor = [NSSortDescriptor sortDescriptorWithKey:@"name" ascending:YES selector:@selector(localizedStandardCompare:)];
+    fetchRequest.sortDescriptors=@[sortDescriptor];
```
Comment (TequilarTomsk) [inline hrm_2/EmployeeListViewController.m]: add " "

**Pull request merged**